addons/mail/models/res_users.py

```diff
@@ -102,6 +102,10 @@ def unlink(self):
         self._unsubscribe_from_channels()
         return super().unlink()
 
+    @api.model
+    def _get_public_users(self):
```

Comment:
This is globally wrong (public users are any users in public groups), so it should probably not be defined on users model but only in mail if specific to some of your flow.

Reply:
Thank you for the check, better with this rename?

Note: currently in `mail` and `website` due to the presence of the key here for now, but it could be moved to `im_livechat` and `website_livechat` in master (and eventually removed the day we make livechat use guest).

Reply:
Actually, why do you need those ? Can't you compute their share/public status directly in mail ? I suppose this is linked to guests / anonymous ? It would allow to stop sending information in the wild ...

Side question: why do we have tons of public users ? My remark is that standard should not be guided by strange internal practices ...

Reply:
Not clear I think. I suppose you want to know if recipients / operators / ... are public ? Can't it be computed with recipients, instead of sending a list of public partners ?